File: AndroidApp/guis_java/app/src/main/java/com/example/loginin/CanvasActivity.java
package com.example.loginin;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.Rect;
import android.os.Bundle;
import android.view.MotionEvent;
import android.view.SurfaceHolder;
import android.view.SurfaceView;
import android.view.View;

import com.example.loginin.R;

import java.util.ArrayList;

public class CanvasActivity extends Activity {

    class Circle{
        float x;
        float y;
        float r;

        public Circle(float x, float y, float r){
            this.x = x;
            this.y = y;
            this.r = r;
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(new DrawView(this));
    }

    class DrawView extends SurfaceView {

        ArrayList<Circle> db = new ArrayList<>();
        SurfaceHolder surfaceHolder;
        Paint p;
        private float r = 60;

        public DrawView(Context context) {
            super(context);
            surfaceHolder = getHolder();
            p = new Paint();
            p.setColor(Color.BLACK);
            p.setStyle(Paint.Style.FILL);
        }

        @Override
        public boolean onTouchEvent(MotionEvent event) {
            float x = event.getX();
            float y = event.getY();
            System.out.println("x: " + x + ", y: " + y);
            Circle cir = new Circle(x, y, r);

            switch (event.getAction()) {
                case MotionEvent.ACTION_DOWN:
                    System.out.println("down");
                    if (x < 150 && y > 2500){
                        if (db.size() != 0)
                            db.remove(db.size() - 1);
                        return true;
                    }
                    db.add(cir);
                    break;
                case MotionEvent.ACTION_MOVE:
                    System.out.println("move");
                    break;
                case MotionEvent.ACTION_UP:
                    if (surfaceHolder.getSurface().isValid()) {
                        Canvas canvas = surfaceHolder.lockCanvas();
                        canvas.drawColor(Color.WHITE);
                        canvas.drawRect(0, 2500, 150, 2700, p);
                        for (Circle crl:db) {
                            canvas.drawCircle(crl.x, crl.y, crl.r, p);
                        }
                        surfaceHolder.unlockCanvasAndPost(canvas);
                    }
                    break;
                case MotionEvent.ACTION_CANCEL:
                    System.out.println("cancel");
                    break;
            }

            return true;
        }

    }
}

File: AndroidApp/guis_kotlin/app/src/main/java/com/example/guis_kotlin/ui/login/CanvasActivity.kt
package com.example.guis_kotlin.ui.login

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.os.Bundle
import android.view.MotionEvent
import android.view.SurfaceHolder
import android.view.SurfaceView
import androidx.appcompat.app.AppCompatActivity
import com.example.guis_kotlin.R

class CanvasActivity : AppCompatActivity() {

    class Circle(val x: Float, val y: Float, val r: Float) {}

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(DrawView(this))
    }


    class DrawView(context: Context) : SurfaceView(context) {
        var db = ArrayList<Circle>()
        var surfaceHolder: SurfaceHolder = holder
        var p: Paint = Paint()
        private var r: Float = 60F

        init {
            p.color = Color.BLACK
            p.style = Paint.Style.FILL
        }

        override fun onTouchEvent(event: MotionEvent): Boolean {
            val x = event.x
            val y = event.y
            println("x: $x, y: $y")
            val cir = Circle(x, y, 60F)
            when (event.action){
                MotionEvent.ACTION_DOWN -> {
                    println("down")
                    if (x < 50 && y < 50){
                        if (db.size != 0)
                            db.removeAt(db.size - 1)
                        return true
                    }
                    db.add(cir)
                }
                MotionEvent.ACTION_UP ->{
                    println("up")
                    if (surfaceHolder.surface.isValid){
                        val canvas: Canvas = surfaceHolder.lockCanvas()
                        canvas.drawColor(Color.WHITE)
                        canvas.drawRect(0F, 0F, 50F, 50F, p)
                        db.forEach{ crl ->
                            canvas.drawCircle(crl.x, crl.y, crl.r, p)
                        }
                        surfaceHolder.unlockCanvasAndPost(canvas)
                    }
                }
                MotionEvent.ACTION_MOVE ->{
                    println("move")
                }
                MotionEvent.ACTION_CANCEL ->{
                    println("cancel")
                }
            }
            return true
        }

    }
}
